dot/parachain/candidate-validation/worker.go

```diff
@@ -22,10 +23,26 @@ type workerTask struct {
 	timeoutKind      parachaintypes.PvfExecTimeoutKind
 }
 
-func newWorker(validationCode parachaintypes.ValidationCode) (*worker, error) {
-	parachainRuntime, err := parachainruntime.SetupVM(validationCode)
-	if err != nil {
-		return nil, err
+var ErrorPreCheckTimeout = errors.New("precheck timed out")
+
+func newWorker(validationCode parachaintypes.ValidationCode, setupTimeout time.Duration) (*worker, error) {
+	runtimeSetupResultCh := make(chan *resultWithError)
+	var parachainRuntime *parachainruntime.Instance
+	go func() {
+		rt, err := parachainruntime.SetupVM(validationCode)
+		runtimeSetupResultCh <- &resultWithError{result: rt, err: err}
+	}()
+
+	select {
+	case validationResultWErr := <-runtimeSetupResultCh:
+		if validationResultWErr.err != nil {
+			logger.Errorf("setting up runtime instance: %w", validationResultWErr.err)
+			return nil, validationResultWErr.err
```

Comment:
the returned error will be logged eventually. So why do you log in here? looks redundant to me.

Reply:
Good point, I've removed this.